spark-connector/src/main/java/com/datastrato/gravitino/spark/connector/iceberg/IcebergAdaptor.java

```diff
@@ -0,0 +1,58 @@
+/*
+ *  Copyright 2024 Datastrato Pvt Ltd.
+ *  This software is licensed under the Apache License version 2.
+ */
+
+package com.datastrato.gravitino.spark.connector.iceberg;
+
+import com.datastrato.gravitino.rel.Table;
+import com.datastrato.gravitino.spark.connector.GravitinoCatalogAdaptor;
+import com.datastrato.gravitino.spark.connector.GravitinoSparkConfig;
+import com.datastrato.gravitino.spark.connector.PropertiesConverter;
+import com.datastrato.gravitino.spark.connector.table.SparkBaseTable;
+import com.google.common.base.Preconditions;
+import java.util.HashMap;
+import java.util.Map;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.iceberg.spark.SparkCatalog;
+import org.apache.spark.sql.connector.catalog.Identifier;
+import org.apache.spark.sql.connector.catalog.TableCatalog;
+import org.apache.spark.sql.util.CaseInsensitiveStringMap;
+
+/** IcebergAdaptor provides specific operations for Iceberg Catalog to adapt to GravitinoCatalog. */
+public class IcebergAdaptor implements GravitinoCatalogAdaptor {
+
+  @Override
+  public PropertiesConverter getPropertiesConverter() {
+    return new IcebergPropertiesConverter();
+  }
+
+  @Override
+  public SparkBaseTable createSparkTable(
+      Identifier identifier,
+      Table gravitinoTable,
+      TableCatalog sparkCatalog,
+      PropertiesConverter propertiesConverter) {
+    return new SparkIcebergTable(identifier, gravitinoTable, sparkCatalog, propertiesConverter);
+  }
+
+  @Override
+  public TableCatalog createAndInitSparkCatalog(
+      String name, CaseInsensitiveStringMap options, Map<String, String> properties) {
+    Preconditions.checkArgument(
```

Comment:
we'd better support all Gravitino Iceberg catalog backends: hive, jdbc, and  memory.

Reply:
`SparkCatalog of Iceberg` can proxy `HiveCatalog`, `JdbcCatalog` and `RestCatalog`.   In the future, maybe we should support `RestCatalog`.   Perhaps `SparkCatalog` is a better choice.   If we use `RestCatalog`,  users can not care about the backends.

Reply:
SparkCatalog: https://github.com/apache/iceberg/blob/857590f7fa8b04212e9ace2efe5f1378ede4f96f/spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/SparkCatalog.java#L148-L154

https://github.com/apache/iceberg/blob/main/core/src/main/java/org/apache/iceberg/CatalogUtil.java#L272-L310

Reply:
supporting `RestCatalog` backend seems another issue. we could support it if nesseseary.

Reply:
ok, i will trace it in another issue.